templates/index-public.php

```diff
@@ -26,6 +26,16 @@
 		'init',
 	]
 );
+
+// The 'application' property is set by \OC\TemplateLayout when the template is
+// rendered as 'user'. As this is a public page the template is rendered as
+// 'base', and thus the application name must be explicitly set by the template.
+foreach(\OC_App::getNavigation() as $entry) {
+	if ($entry['active']) {
+		$_['application'] = $entry['name'];
```

Comment:
Let's instead hardcode this. No need to loop over all navigation entries, when we display `$l->t('Talk')` only anyway.